src/pybind/mgr/dashboard/frontend/src/app/shared/api/smb.service.ts

```diff
@@ -19,13 +26,52 @@ export class SmbService {
   baseURL = 'api/smb';
   private modalDataSubject = new Subject<DomainSettings>();
   modalData$ = this.modalDataSubject.asObservable();
+  dataUploader = signal<SMBResource | null>(null);
+  dataUploaderHider = signal<boolean>(false);
 
-  constructor(private http: HttpClient) {}
+  constructor(private http: HttpClient, private notificationService: NotificationService) {}
 
   passData(data: DomainSettings) {
     this.modalDataSubject.next(data);
   }
 
+  setDataUploaded(data: SMBResource): void {
+    this.dataUploader.set(data);
+  }
+
+  uploadData(event: any) {
+    const file = event.values().next().value?.file;
+    if (!file) {
+      return;
+    }
+
+    if (!(file.type === APPYAML || file.type === APPJSON)) {
+      this.notificationService.show(
+        NotificationType.error,
+        $localize`Invalid file type: only .json or .yaml accepted`
+      );
```

Comment:
i don't see this validations being raised though?

Reply:
Not sure why to be honest, I can see it whenever I use other file types

[invalid_file.webm](https://github.com/user-attachments/assets/6d2a71cf-56d5-4422-a399-17c4e774e346)

Reply:
I think there was an issue with my cluster then.. I didn't see any notification (not just on this PR but in general). It looked like the toasts handler went on a strike.